### ISSUE TITLE
Helm dependancy update is not locking version in Chart.lock if repo is not added into repo list

### ISSUE THREAD

**Issue by GoaMind** [helm/helm#8663]
`helm dependancy update` is not locking version in `Chart.lock`
If in `Chart.yaml` version specified like that:
```
  version: ~0.3.0
  # or version: 0.3.*
```
after `helm dep up`, in `Chart.lock` file it will appear like that:
```
  version: ~0.3.0
  # or version: 0.3.*
```
If remote repo is not added to Helm repo list

---------------

👍   IT WORKS (locking version properly)
if repository is local path via `file:///`
if repository is remote path  `https://some-registry` and it is added to `helm repo list`
in `Chart.lock` will be locked specific version: `version: 0.3.29`

👎   IT DOES NOT WORK (not locking version)
if repository is remote path `https://some-registry` and it is NOT added to `helm repo list`
in `Chart.lock` will be locked pattern version: `version: 0.3.*` or `version: ~0.3.0`

Output of `helm version`:
```
version.BuildInfo{Version:"v3.3.0", GitCommit:"8a4aeec08d67a7b84472007529e8097ec3742105", GitTreeState:"dirty", GoVersion:"go1.14.6"}
```

Output of `kubectl version`:
```
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.3", GitCommit:"2e7996e3e2712684bc73f0dec0200d64eec7fe40", GitTreeState:"clean", BuildDate:"2020-05-21T14:50:54Z", GoVersion:"go1.14.3", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"16+", GitVersion:"v1.16.6-beta.0", GitCommit:"e7f962ba86f4ce7033828210ca3556393c377bcc", GitTreeState:"clean", BuildDate:"2020-01-15T08:18:29Z", GoVersion:"go1.13.5", Compiler:"gc", Platform:"linux/amd64"}
```

Cloud Provider/Platform (AKS, GKE, Minikube etc.): 
Not related / done locally

**Comment by mattfarina** [comment]
@GoaMind This is a bug we know about. It was fixed in #8478 which is scheduled to go out as part o v3.4.0. Can you confirm this is fixed in the latest unreleased code?

**Comment by GoaMind** [comment]
Confirm, tested with latest `master` branch and I was not able to reproduce this bug anymore.
Thank you!

Output of `helm version`:
```
version.BuildInfo{Version:"v3.3+unreleased", GitCommit:"4f7b9fcb67424008a839845397838b491036ca61", GitTreeState:"clean", GoVersion:"go1.15"}
```

**Comment by hickeyma** [comment]
Closing as fixed in #8478 and confirmed by OP.
